File: 2023/internal/day_01/day_01_test.go
package day_01

import (
	_ "embed"
	"testing"

	"github.com/stretchr/testify/require"
)

//go:embed part1_sample_input.txt
var part1_sample_input string
//go:embed part2_sample_input.txt
var part2_sample_input string


func TestPart1(t *testing.T) {
	require := require.New(t)

	// test on smaple input
	require.Equal(142, int(Part1(part1_sample_input)))

	// test on real input
	require.Equal(56397, int(Part1(input)))
}


func TestPart2(t *testing.T) {
	require := require.New(t)

	//test on sample input
	require.Equal(281, Part2(part2_sample_input))

	// test on real input
	require.Equal(55701, Part2(input))
}

File: 2024/src/day01/day01.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { Day01Solution } from "./day01.ts";


describe("Day01Solution", () => {
	const solution = new Day01Solution();

	const sampleInput = [
		"3   4\n",
		"4   3\n",
		"2   5\n",
		"1   3\n",
		"3   9\n",
		"3   3\n",
	];

	const realInput = solution.parseInput(Deno.readTextFileSync("src/day01/input.txt"));

	describe("part1()", () => {

		it("should return the correect value for the sample input", () => {
			expect(solution.part1(sampleInput)).toBe(11);
		});

		it("should return the correct value for the real input", () => {
			expect(solution.part1(realInput)).toBe(2285373);
		});

	});

	describe("part2()", () => {
		
		it("should return the correct value for the sample input", () => {
			expect(solution.part2(sampleInput)).toBe(31);
		});

		it("should return the correct value for the real input", () => {
			expect(solution.part2(realInput)).toBe(21142653);
		});

	});

});
